velox/exec/tests/TableScanTest.cpp

```diff
@@ -1508,6 +1514,34 @@ TEST_F(TableScanTest, partitionedTableDoubleKey) {
   testPartitionedTable(filePath->path, DOUBLE(), "3.5");
 }
 
+TEST_F(TableScanTest, partitionedTableDecimalKey) {
+  auto rowType = ROW({"c0", "c1"}, {BIGINT(), DOUBLE()});
+  auto vectors = makeVectors(10, 1'000, rowType);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+  createDuckDbTable(vectors);
+  testPartitionedTable(
+      filePath->path,
+      DECIMAL(12, 3),
+      "123456789123",
+      "cast('123456789.123' as decimal(12, 3))");
```

Comment:
There is quite a bit of repetition here. Both value and type are repeated. Can we generate cast SQL automatically? For example,

```
std::string makeCastSql(const variant& v, const TypePtr& type) {
  std::ostringstream out;
  out << "CAST('" << v.toJson(type) << "' AS ";
  toTypeSql(type, out);
  out << ")";
  return out.str();
}

TEST_F(VectorTest, xxx) {
  {
    TypePtr t = DECIMAL(12, 3);
    int64_t n = 123456789123;
    variant v(n);

    LOG(ERROR) << makeCastSql(v, t);
  }

  {
    TypePtr t = DECIMAL(36, 18);
    int128_t n = HugeInt::parse("123456789123456789123456789123456789");
    variant v(n);

    LOG(ERROR) << makeCastSql(v, t);
  }

  FAIL();
}

VectorTest.cpp:3665] CAST('123456789.123' AS DECIMAL(12, 3))
VectorTest.cpp:3673] CAST('123456789123456789.123456789123456789' AS DECIMAL(36, 18))

```

Reply:
Thanks for your nice suggestion. Do you think we should generate cast for all types covered by `testPartitionedTable`, or only for decimal specially? Thanks.

Reply:
@rui-mo I don't have a strong opinion. If this can be applies to all types, the code will be simpler.

Reply:
Got it. Appreciate your help. I will try to apply to all types.

Reply:
@rui-mo Thank you, Rui.

Reply:
Updated this PR to apply cast for all types. Could you help review again? Thanks.